From 503a0b8da8ed299a00ac95fa24d1fc37c3e831d3 Mon Sep 17 00:00:00 2001
From: Herbert Xu <herbert@gondor.apana.org.au>
Date: Wed, 29 Mar 2006 07:35:34 +1100
Subject: [PATCH] [PARSER] Removed useless parsebackquote flag

The parsebackquote flag is only used in a test where it always has the
value zero.  So we can remove it altogether.
---
 ChangeLog    |    4 ++++
 src/parser.c |    8 +-------
 2 files changed, 5 insertions(+), 7 deletions(-)

diff --git a/ChangeLog b/ChangeLog
index 02b966c..5dd6d40 100644
--- a/ChangeLog
+++ b/ChangeLog
@@ -1,3 +1,7 @@
+2006-03-29  Herbert Xu <herbert@gondor.apana.org.au>
+
+	* Removed useless parsebackquote flag.
+
 2006-01-12  Herbert Xu <herbert@gondor.apana.org.au>
 
 	* Fixed eval exit status with empty arguments.
diff --git a/src/parser.c b/src/parser.c
index c62a950..375fd54 100644
--- a/src/parser.c
+++ b/src/parser.c
@@ -76,7 +76,6 @@ struct heredoc {
 
 
 struct heredoc *heredoclist;	/* list of here documents to read */
-int parsebackquote;		/* nonzero if we are inside backquotes */
 int doprompt;			/* if set, prompt the user */
 int needprompt;			/* true if interactive and at start of line */
 int lasttoken;			/* last token read */
@@ -1019,7 +1018,7 @@ quotemark:
 endword:
 	if (syntax == ARISYNTAX)
 		synerror("Missing '))'");
-	if (syntax != BASESYNTAX && ! parsebackquote && eofmark == NULL)
+	if (syntax != BASESYNTAX && eofmark == NULL)
 		synerror("Unterminated quoted string");
 	if (varnest != 0) {
 		startlinno = plinno;
@@ -1263,7 +1262,6 @@ badsub:			synerror("Bad substitution");
 
 parsebackq: {
 	struct nodelist **nlpp;
-	int savepbq;
 	union node *n;
 	char *volatile str;
 	struct jmploc jmploc;
@@ -1274,11 +1272,9 @@ #ifdef __GNUC__
 	(void) &saveprompt;
 #endif
 
-	savepbq = parsebackquote;
 	if (setjmp(jmploc.loc)) {
 		if (str)
 			ckfree(str);
-		parsebackquote = 0;
 		handler = savehandler;
 		longjmp(handler->loc, 1);
 	}
@@ -1360,7 +1356,6 @@ done:
 		nlpp = &(*nlpp)->next;
 	*nlpp = (struct nodelist *)stalloc(sizeof (struct nodelist));
 	(*nlpp)->next = NULL;
-	parsebackquote = oldstyle;
 
 	if (oldstyle) {
 		saveprompt = doprompt;
@@ -1396,7 +1391,6 @@ done:
 		str = NULL;
 		INTON;
 	}
-	parsebackquote = savepbq;
 	handler = savehandler;
 	if (arinest || dblquote)
 		USTPUTC(CTLBACKQ | CTLQUOTE, out);
-- 
1.4.3.1

